src/connectors/autocomplete/connectAutocomplete.ts

```diff
@@ -102,24 +102,36 @@ search.addWidgets([
     return {
       $$type: 'ais.autocomplete',
 
-      init({ instantSearchInstance, helper }) {
+      init(initOptions) {
+        const { helper, renderState, instantSearchInstance } = initOptions;
         connectorState.refine = (query: string) => {
           helper.setQuery(query).search();
         };
 
         renderFn(
           {
-            widgetParams,
-            currentRefinement: helper.state.query || '',
-            indices: [],
-            refine: connectorState.refine,
+            ...this.getWidgetRenderState!(renderState, initOptions)
+              .autocomplete!,
             instantSearchInstance,
           },
           true
         );
       },
 
-      render({ helper, scopedResults, instantSearchInstance }) {
+      render(renderOptions) {
+        const { renderState, instantSearchInstance } = renderOptions;
+
+        renderFn(
+          {
+            ...this.getWidgetRenderState!(renderState, renderOptions)
+              .autocomplete!,
+            instantSearchInstance,
+          },
+          false
+        );
+      },
+
+      getWidgetRenderState(renderState, { helper, scopedResults }) {
         const indices = scopedResults.map(scopedResult => {
```

Comment:
I think scopedResults might be undefined in init (or does it get called with [])?

Reply:
It's `[]` in init.